spec/operations/source_spec.rb

```diff
@@ -2,33 +2,158 @@
 require "topological_inventory/ansible_tower/operations/source"
 
 RSpec.describe(TopologicalInventory::AnsibleTower::Operations::Source) do
-  describe "availability_check" do
-    let(:host_url) { "https://cloud.redhat.com" }
-    let(:external_tenant) { "11001" }
-    let(:identity) do
-      { "x-rh-identity" => Base64.strict_encode64({ "identity" => { "account_number" => external_tenant, "user" => { "is_org_admin" => true } } }.to_json) }
-    end
-    let(:headers) { {"Content-Type" => "application/json"}.merge(identity) }
-
-    it "makes a patch request to update the availability_status of a source" do
-      source_id = "201"
-      payload =
-        {
-          "params" => {
-            "source_id"       => source_id,
-            "external_tenant" => external_tenant,
-            "timestamp"       => Time.now.utc
-          }
+  describe 'availability_check' do
+    let(:host_url) { 'https://cloud.redhat.com' }
+    let(:sources_api_path) { '/api/sources/v3.0' }
+    let(:sources_internal_api_path) { '/internal/v1.0' }
+    let(:sources_api_url) { "#{host_url}#{sources_api_path}" }
+
+    let(:external_tenant) { '11001' }
+    let(:identity) { {'x-rh-identity' => Base64.strict_encode64({'identity' => {'account_number' => external_tenant, 'user' => {'is_org_admin' => true}}}.to_json)} }
+    let(:headers) { {'Content-Type' => 'application/json'}.merge(identity) }
+    let(:source_id) { '123' }
+    let(:endpoint_id) { '234' }
+    let(:authentication_id) { '345' }
+    let(:payload) do
+      {
+        'params' => {
+          'source_id'       => source_id,
+          'external_tenant' => external_tenant,
+          'timestamp'       => Time.now.utc
         }
+      }
+    end
+
+    let(:list_endpoints_response) { "{\"meta\":{\"count\":1,\"limit\":100,\"offset\":0},\"links\":{\"first\":\"#{sources_api_path}/sources/#{source_id}/endpoints?offset=0\",\"last\":\"#{sources_api_path}/sources/#{source_id}/endpoints?offset=0\"},\"data\":[{\"default\":true,\"host\":\"10.0.0.1\",\"id\":\"#{endpoint_id}\",\"path\":\"/\",\"role\":\"ansible\",\"scheme\":\"https\",\"source_id\":\"#{source_id}\",\"tenant\":\"#{external_tenant}\"}]}"}
```

Comment:
I think on these we can remove the `meta` fields - the client will still parse the object fine and it makes it easier to read since we're only inserting the fields we need in the response.